src/pow.cpp

```diff
@@ -355,12 +355,16 @@ void multinomial_coefficients_mpz(int m, int n, map_vec_mpz &r)
 RCP<const Basic> pow_expand(const RCP<const Pow> &self)
 {
     RCP<const Basic> _base = expand(self->base_);
+    bool negative_pow = false;
     if (! is_a<Integer>(*self->exp_) || ! is_a<Add>(*_base))
         return self;
```

Comment:
These two lines ensure that `_base` is always `Add`, so I think we can speed it up by using your original approach with `rcp(new Pow(...))`. And put an assert statement above it ---- that way we make sure the expression is what we think it is, and if so, pass it directly to Pow.